src/calva-fmt/src/config.ts

```diff
@@ -26,9 +27,25 @@ function configuration(
     };
 }
 
-function readConfiguration() {
+async function readConfiguration(): Promise<{
+    'format-as-you-type': boolean;
+    'keep-comment-forms-trail-paren-on-own-line?': boolean;
+    'cljfmt-options-string': string;
+    'cljfmt-options': object;
+}> {
     const workspaceConfig = vscode.workspace.getConfiguration('calva.fmt');
     const configPath: string = workspaceConfig.get('configPath');
+    if (configPath === LSP_CONFIG_KEY && !lspFormatConfigFetched) {
+        lspFormatConfig = await lsp
+            .getCljFmtConfig()
+            .then((c) => {
+                lspFormatConfigFetched = true;
+                return c;
+            })
+            .catch((e) => {
+                lspFormatConfigFetched = true;
+            });
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**🎯  Suggestion:** Maybe a `try catch` here is easier to read.‏

```suggestion
      try {
        lspFormatConfig = await lsp.getCljFmtConfig(); 
        lspFormatConfigFetched = true;
      } catch (_) {
        lspFormatConfigFetched = true;
      }
```